### DISCUSSION TITLE
Potential AWS API Limit Exceeding

### DISCUSSION THREAD

**Discussion by ramses999** [prowler-cloud/prowler#2807] [Q&A]
I want to run prowler every hour (or more often) with all checks, but I don’t understand how this can affect the overall infrastructure.
There is an assumption that such frequent launches may exceed any API throttling limits and this will potentially break production.

Tell me, has anyone already studied this topic? does AWS have similar limits? how many API requests does Prowler generate with all the checks?

**Comment by ramses999** [comment]
For example, I encountered a problem that when the “iam_root_mfa_enabled” check is often called, it calls “Get Credential Report” and after several dozen attempts it just hangs

<img width="962" alt="CLI" src="https://github.com/prowler-cloud/prowler/assets/108680000/206e4286-cacd-4f19-8719-694bd95e5a4c">
<img width="962" alt="UI" src="https://github.com/prowler-cloud/prowler/assets/108680000/a493541d-0552-4c52-bf94-99473bcca7ff">

**Comment by jfagoagas** [comment]
Hi @ramses999 we don't have an exact count of how many API requests Prowler makes but you can use the flag `--aws-retries-max-attempts` to set the maximum attempts for the Boto3 standard retrier config.

Regarding the IAM Credential Report, it is generated by IAM and is stored 4 hours so you will get the same result in each 4-hour window. It is strange the behaviour you noticed, if so please open an issue and share the logs for us to review.
